### PR TITLE
Use multi-selection api in workbench

### PR DESCRIPTION
**Pull request by gcoutable**
### Type of this PR 

- [ ] Bug fix
- [x] New feature or improvement
- [ ] Documentation
- [ ] Cleanup
- [ ] Test
- [ ] Build/Releng

### Issue(s)

https://github.com/eclipse-sirius/sirius-components/issues/134

### What does this PR do?
This commit does not add behavior, it only prepares the workbench and
all workbench components using the selection to support the multiple
selection.

### CODE REVIEW

File: frontend/src/onboarding/NewRepresentationArea.tsx [hunk 1]
```diff
@@ -121,8 +122,8 @@ export const NewRepresentationArea = ({
     return <AreaContainer title={title} subtitle="You need edit access to create representations" />;
   } else {
     let subtitle =
-      selection && representationDescriptions.length > 0
-        ? 'Select the representation to create on ' + selection.label
+      selectedItem && representationDescriptions.length > 0
+        ? 'Select the representation to create on ' + selection.entries[0].label
```
Comment (sbegaudeau) [inline frontend/src/onboarding/NewRepresentationArea.tsx:126]: `selectedItem.label` to stay coherent but it works

File: frontend/src/workbench/Workbench.tsx [hunk 2]
```diff
@@ -62,8 +62,14 @@ export const Workbench = ({
   const { selection, representations, displayedRepresentation } = context;
 
   const setSelection = (selection: Selection) => {
-    const isRepresentation = registry.isRepresentation(selection.kind);
-    const updateSelectionEvent: UpdateSelectionEvent = { type: 'UPDATE_SELECTION', selection, isRepresentation };
+    const representations: Representation[] = selection?.entries.filter((entry) =>
```
Comment (sbegaudeau) [inline frontend/src/workbench/Workbench.tsx:65]: no need for `?` the selection cannot be null here, can it?

File: frontend/src/workbench/WorkbenchMachine.ts [hunk 3]
```diff
@@ -20,21 +20,25 @@ export interface WorkbenchStateSchema {
...
+  representations: Representation[];
+};
 export type WorkbenchEvent = UpdateSelectionEvent | ShowRepresentationEvent | HideRepresentationEvent;
 
 export const workbenchMachine = Machine<WorkbenchContext, WorkbenchStateSchema, WorkbenchEvent>(
   {
     initial: 'initial',
     context: {
-      selection: null,
+      selection: { entries: [] },
```
Comment (sbegaudeau) [inline frontend/src/workbench/WorkbenchMachine.ts:41]: Now that the object `selection` on which everything is connected cannot be `null` anymore, there should not be any occurrences of `selection?` in the code again. There are still some in your PR.

**Review by sbegaudeau — CHANGES_REQUESTED** [review]
You just need to review usages of `selection?` and I'll need to test that with a complete product like Sirius Web but otherwise everything seems ok. Could you please update the `CHANGELOG` too? Thanks

File: frontend/src/onboarding/NewRepresentationArea.tsx (hunk 1 above)
Comment (gcoutable) [reply frontend/src/onboarding/NewRepresentationArea.tsx:126]: Done

File: frontend/src/workbench/Workbench.tsx (hunk 2 above)
Comment (gcoutable) [reply frontend/src/workbench/Workbench.tsx:65]: It should not. Removed.

File: frontend/src/workbench/WorkbenchMachine.ts (hunk 3 above)
Comment (gcoutable) [reply frontend/src/workbench/WorkbenchMachine.ts:41]: Done

**Comment by sbegaudeau** [comment]
I'll need to review the impact on consumers like Sirius Web and add an entry in the CHANGELOG but it looks good.

**Comment by sbegaudeau** [comment]
There seems to have a very minor change in behavior in the lifecycle of the details view. I'll investigate.

File: frontend/src/properties/PropertiesWebSocketContainer.tsx [hunk 4]
```diff
@@ -88,9 +88,13 @@ export const PropertiesWebSocketContainer = ({
...
-    if (currentSelection?.id !== selection?.id) {
-      const isRepresentation = registry.isRepresentation(selection.kind);
-      const switchSelectionEvent: SwitchSelectionEvent = { type: 'SWITCH_SELECTION', selection, isRepresentation };
+    if (selection.entries.length > 0 && selection.entries[0].id !== currentSelection?.id) {
+      const isRepresentation = registry.isRepresentation(selection.entries[0].kind);
+      const switchSelectionEvent: SwitchSelectionEvent = {
+        type: 'SWITCH_SELECTION',
+        selection: selection.entries[0],
+        isRepresentation,
+      };
```
Comment (sbegaudeau) [inline frontend/src/properties/PropertiesWebSocketContainer.tsx:97]: The current selection of the PropertiesWebSocketContainer cannot be null anymore so it does not have the same behavior when there is no selection

**Comment by sbegaudeau** [comment]
I'll close the PR since it is not made against `master` and as such it is not relevant. We will recreate the PR from scratch and merge it.

**Pull request closed without merging**